src/main/scala/memeid/UUID.scala

```diff
@@ -46,29 +82,79 @@ final class UUID private[memeid] (private[memeid] val juuid: JUUID) {
 
 object UUID {
 
-  // The `null` UUID
-  val empty: UUID = new UUID(new JUUID(0, 0))
-
-  /* Constructors */
-
-  def from(msb: Long, lsb: Long): UUID =
-    new UUID(new JUUID(msb, lsb))
-
-  /* Typeclass instances */
-
-  implicit val UUIDHashOrderInstance: Order[UUID] with Hash[UUID] =
-    new Order[UUID] with Hash[UUID] {
+  /**
+   * The nil UUID is special form of UUID that is specified to have all 128 bits set to zero.
+   *
+   * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.7]]
+   */
+  case object Nil extends UUID { override val juuid: JUUID = new JUUID(0, 0) }
+
+  /**
+   * Version 1 UUIDs are those generated using a timestamp and the MAC address of the
+   * computer on which it was generated.
+   *
+   * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.3]]
+   */
+  final class V1 private[memeid] (override private[memeid] val juuid: JUUID) extends UUID
```

Comment:
What's the `class V1 private[memeid]` effect? . Does it make `V1` private outside `memeid` package?

Reply:
Nope, it makes `V1` constructor private outside `memeid`, so no one (but us) can create `V1` UUIDs (to force users to use our convenient methods)

Reply:
Amazing!. Thanks!